src/guide/parameter-binding.md

```diff
@@ -0,0 +1,150 @@
+# Parameter Binding
+
+**Level: Intermediate** 🍃🍃
+
+Starting in .NET 6, route handlers (the Delegate parameters in [EndpointRouteBuilderExtensions](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.builder.endpointroutebuilderextensions) methods) can accept more than just an `HttpContext` as a parameter. Route handlers can now accept any number of parameters with various types bound from different sources. This guide describes the conventions that determine how each parameter is populated.
+
+## Conventions
+
+```cs
+app.MapPut("/todos/{id}", async (TodoDb db, TodoItem updateTodo, int id) =>
+{
+    // ...
+});
+```
+
+The route handler above accepts three parameters:
+
+1. `TodoDb db` which comes from the service provider.
+2. `TodoItem updateTodo` which is read as JSON from the request body.
+3. `int id` which is read from the `{id}` segment of the route.
+
+These were all determined by convention but could be specified explicitly with attributes as follows:
+
+```cs
+using Microsoft.AspNetCore.Mvc;
+
+// ...
+
+app.MapPut("/todos/{id}", async (
+    [FromService] TodoDb db,
+    [FromBody] TodoItem updateTodo,
+    [FromRoute(Name = "id")] int nameDoesNotMatter) =>
+{
+    // ...
+});
+```
+
+Parameters sources are determined using the following rules applied in order:
+
+1. [Parameter attributes](#attributes) take precedence over conventions if they are present.
+2. Any [well-known types](#well-known-types) are bound from the the [HttpContext](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.http.httpcontext) or one of its properties.
+3. `string` parameters are bound from `HttpContext.RouteValues[{ParameterName}]` or `HttpContext.Query[{ParameterName}]` depending on whether `{ParameterName}` is part of the route pattern.
```

Comment:
```suggestion
3. `string` parameters are bound from `HttpContext.RouteValues[{ParameterName}]` `{ParameterName}` is part of the route pattern or `HttpContext.Query[{ParameterName}]` otherwise.
```

nit: feels less ambiguous.